reconcile/glitchtip/reconciler.py

```diff
@@ -0,0 +1,278 @@
+import logging
+from typing import Iterable, Sequence
+
+from reconcile.utils.glitchtip import GlitchtipClient, Organization, Project, Team, User
+
+
+class GlitchtipReconciler:
+    def __init__(self, client: GlitchtipClient, dry_run: bool):
+        self.client = client
+        self.dry_run = dry_run
+
+    def _reconcile_projects(
+        self,
+        organization_slug: str,
+        organization_teams: Sequence[Team],
+        current_projects: Iterable[Project],
+        desired_projects: Iterable[Project],
+    ) -> list[Project]:
+        """Reconcile organization projects."""
+        organization_projects = list(current_projects)
+        for project in set(current_projects).difference(desired_projects):
+            logging.info(
+                ["delete_project", organization_slug, project.slug, self.client.host]
+            )
+            del organization_projects[organization_projects.index(project)]
+            if not self.dry_run:
+                self.client.delete_project(
```

Comment:
What if deleting a project fails, will the rest of the function run?

Reply:
Do you mean to have a `try/except` around?

Reply:
Yes, catching errors, that are "okay" to happen. Wonder if there are more errors, like adding/deleting single users, for instance, that should not cause the integrations to crash. 
Guess there is some dependency between the data that must be solved.

Reply:
I keep it as it is because otherwise, it'll hide errors. I think it's better to have a failing integration instead of error messages somewhere that nobody reads. In this case, if the project deletion fails, then most properly, all other HTTP requests will fail too.